### ISSUE TITLE
Loss of Repeater Field Data when Rearranging Layouts in Flexible Content

### ISSUE THREAD

**Issue by Mytraas** [AdvancedCustomFields/acf#879]
**Describe the bug**
When using a Flexible Content field in ACF, which contains layouts that include Repeater fields, there is an issue where changing the order of the layouts in the WordPress admin interface results in the loss of data from the Repeater field within those layouts.

**To Reproduce**
Create a Flexible Content field in ACF and add layouts that contain Repeater fields.
Populate data in the Repeater fields within the layouts.
In the WordPress admin interface, rearrange the order of the layouts within the Flexible Content field.
Save the changes.
Notice that the data in the Repeater fields has been erased.

**Expected behavior**
Expectations: We expect that rearranging the layouts within the Flexible Content field should not result in data loss within the Repeater fields.

**Statement**
I wanted to provide an update regarding the configuration change I made in PHP. As recommended in the discussion thread found here: [Thread Link](https://support.advancedcustomfields.com/forums/topic/updating-flexible-content-field-makes-rows-disappear-in-backend/), I adjusted the max_input_vars to 10000.

Unfortunately, despite making this adjustment, the issue of data loss when updating Flexible Content fields persists. The rows continue to disappear in the backend as mentioned in the thread.

**Screenshots or Video**
https://github.com/AdvancedCustomFields/acf/assets/33683465/def1c566-4254-410a-b5ce-08cd6d600431

**Version Information:**
- Web Server: Apache/2.4.54 (Unix) OpenSSL/1.0.2u PHP/8.2.0
- PHP Version: 8.2.0
- max_input_vars 10000
- MySQL Server Version: 5.7.39
- WordPress Version: 6.3.1
- ACF Version: 6.2.1.1
 - Browser all

**Additional context**
Thank you for considering this bug report and working towards its resolution. If additional information is required, please feel free to request it.

**Comment by Mytraas** [comment]
This is the JSON file that represents my ACF Fields structure

```
object(ACF_Data)#1038 (4) {
  ["cid"]=>
  string(5) "acf-9"
  ["data"]=>
  array(1) {
    ["group_aee8ef13"]=>
    array(6) {
      ["key"]=>
      string(14) "group_aee8ef13"
      ["title"]=>
      string(14) "Product Layout"
      ["local"]=>
      string(3) "php"
      ["menu_order"]=>
      int(2)
      ["location"]=>
      array(2) {
        [0]=>
        array(1) {
          [0]=>
          array(3) {
            ["param"]=>
            string(9) "post_type"
            ["operator"]=>
            string(2) "=="
            ["value"]=>
            string(7) "product"
          }
        }
        [1]=>
        array(1) {
          [0]=>
          array(3) {
            ["param"]=>
            string(9) "post_type"
            ["operator"]=>
            string(2) "=="
            ["value"]=>
            string(4) "page"
          }
        }
      }
      ["style"]=>
      string(8) "seamless"
    }
  }
  ["aliases"]=>
  array(0) {
  }
  ["multisite"]=>
  bool(false)
}
array(1) {
  [0]=>
  array(16) {
    ["ID"]=>
    int(0)
    ["key"]=>
    string(14) "group_aee8ef13"
    ["title"]=>
    string(14) "Product Layout"
    ["fields"]=>
    array(0) {
    }
    ["location"]=>
    array(2) {
      [0]=>
      array(1) {
        [0]=>
        array(3) {
          ["param"]=>
          string(9) "post_type"
          ["operator"]=>
          string(2) "=="
          ["value"]=>
          string(7) "product"
        }
      }
      [1]=>
      array(1) {
        [0]=>
        array(3) {
          ["param"]=>
          string(9) "post_type"
          ["operator"]=>
          string(2) "=="
          ["value"]=>
          string(4) "page"
        }
      }
    }
    ["menu_order"]=>
    int(2)
    ["position"]=>
    string(6) "normal"
    ["style"]=>
    string(8) "seamless"
    ["label_placement"]=>
    string(3) "top"
    ["instruction_placement"]=>
    string(5) "label"
    ["hide_on_screen"]=>
    array(0) {
    }
    ["active"]=>
    bool(true)
    ["description"]=>
    string(0) ""
    ["show_in_rest"]=>
    bool(false)
    ["local"]=>
    string(3) "php"
    ["_valid"]=>
    bool(true)
  }
}
```

**Comment by amichel86** [comment]
I'm experiencing some what of the same issue with the Flexible Content field, yesterday all my data was there in one of my layouts, today I lost data in some fields in another layout. Fields have the same name but can that be the problem?

https://www.loom.com/share/1e16b5523766437e91c9c3a2339e2e56

**Comment by amichel86** [comment]
> Fields can have the same name as long as they are in different groups. This is literally a bug in ACF, I think.

You're right, I don't know why it wasn't working the first time, but everything's fine now. Thanks!
